webhooks/appmesh/gatewayroute_mutator.go

```diff
@@ -0,0 +1,104 @@
+package appmesh
+
+import (
+	"context"
+	"fmt"
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/mesh"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/virtualgateway"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/webhook"
+	"github.com/pkg/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+const apiPathMutateAppMeshGatewayRoute = "/mutate-appmesh-k8s-aws-v1beta2-gatewayroute"
+
+// NewGatewayRouteMutator returns a mutator for GatewayRoute.
+func NewGatewayRouteMutator(meshMembershipDesignator mesh.MembershipDesignator, virtualGatewayMembershipDesignator virtualgateway.MembershipDesignator) *gatewayRouteMutator {
+	return &gatewayRouteMutator{
+		meshMembershipDesignator:           meshMembershipDesignator,
+		virtualGatewayMembershipDesignator: virtualGatewayMembershipDesignator,
+	}
+}
+
+var _ webhook.Mutator = &gatewayRouteMutator{}
+
+type gatewayRouteMutator struct {
+	meshMembershipDesignator           mesh.MembershipDesignator
+	virtualGatewayMembershipDesignator virtualgateway.MembershipDesignator
+}
+
+func (m *gatewayRouteMutator) Prototype(req admission.Request) (runtime.Object, error) {
+	return &appmesh.GatewayRoute{}, nil
+}
+
+func (m *gatewayRouteMutator) MutateCreate(ctx context.Context, obj runtime.Object) (runtime.Object, error) {
+	gr := obj.(*appmesh.GatewayRoute)
+	mesh, err := m.designateMeshMembership(ctx, gr)
+	if err != nil {
+		return nil, err
+	}
+	virtualGateway, err := m.designateVirtualGatewayMembership(ctx, gr)
+	if err != nil {
+		return nil, err
+	}
+	if virtualGateway.Spec.MeshRef.UID != mesh.UID {
```

Comment:
we can use mesh.isMeshReferenced(mesh, *virtualGateway.Spec.MeshRef) to check for both name & UID.
however need to either rename the "mesh" variable to something like "ms", or rename the mesh package when import.